scaleway/resource_instance_volume.go

```diff
@@ -171,15 +171,6 @@ func resourceScalewayInstanceVolumeUpdate(ctx context.Context, d *schema.Resourc
 		if oldSize, newSize := d.GetChange("size_in_gb"); oldSize.(int) > newSize.(int) {
 			return diag.FromErr(fmt.Errorf("block volumes cannot be resized down"))
 		}
-		_, err := instanceAPI.WaitForVolume(&instance.WaitForVolumeRequest{
```

Comment:
I think we need both. We cannot update if it is not in a stable state

Reply:
Hum, we are in an update context so the volume should be in a stable state. If not the API should reject the call IMO.

Reply:
Maybe another operation coming from the console or the API through the CLI altered the state of the volume. Putting a wait is not a problem, at worst it will do a request, see it is ok and move on.